staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -630,6 +630,36 @@ type PersistentVolumeClaimSpec struct {
 	StorageClassName *string `json:"storageClassName,omitempty" protobuf:"bytes,5,opt,name=storageClassName"`
 }
 
+// PvcConditionType is a valid value of PvcCondition.Type
+type PvcConditionType string
+
+const (
+	// Ready means any pending operation on pvc is successfully completed
+	PvcReady PvcConditionType = "Ready"
+	// An user trigger resize of pvc has been started
+	PvcResizeStarted PvcConditionType = "ResizeStarted"
+	// PvcResizeFailed means user trigged resize of PVC has failed
+	PvcResizeFailed PvcConditionType = "ResizeFailed"
+)
+
+// PvcCondition contails details about state of pvc
+type PvcCondition struct {
+	Type   PvcConditionType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=PvcConditionType"`
+	Status ConditionStatus  `json:"status" protobuf:"bytes,2,opt,name=status,casttype=ConditionStatus"`
+	// Last time we probed the condition.
+	// +optional
+	LastProbeTime metav1.Time `json:"lastProbeTime,omitempty" protobuf:"bytes,3,opt,name=lastProbeTime"`
```

Comment:
If you don't plan on setting LastProbeTime omit it from this struct.  We generally don't set it because it's expensive to calculate.

Reply:
I do plan to keep it for now, because when we implement file system resize,  we want to report on progress of resize. We may have to introduce additional conditions at that point though. But for now, I am leaving it as it is.